pkg/controller/wip/controller.go

```diff
@@ -382,16 +387,30 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) {
 	glog.V(4).Infof("Creating client for Broker %v, URL: %v", broker.Name, broker.Spec.URL)
 	brokerClient := c.brokerClientCreateFunc(broker.Name, broker.Spec.URL, username, password)
 
-	// TODO: parameters
+	parameters, err := unmarshalParameters(instance.Spec.Parameters.Raw)
+	if err != nil {
+		glog.Errorf("Failed to unmarshal Instance parameters\n%s\n %v", instance.Spec.Parameters, err)
+		c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			errorWithParameters,
+			"Error unmarshaling instance parameters",
+		)
+		return
+	}
 
 	request := &brokerapi.CreateServiceInstanceRequest{
-		ServiceID: serviceClass.OSBGUID,
-		PlanID:    servicePlan.OSBGUID,
+		ServiceID:  serviceClass.OSBGUID,
+		PlanID:     servicePlan.OSBGUID,
+		Parameters: parameters,
 	}
 
 	// TODO: handle async provisioning
 
 	glog.V(4).Infof("Provisioning a new Instance %v/%v of ServiceClass %v at Broker %v", instance.Namespace, instance.Name, serviceClass.Name, broker.Name)
+	// TODO(vaikas): Who populates the OSBGUID here? I thought controller creates a GUID, but I don't
+	// see it being generated anywhere...
```

Comment:
We already have a default that makes GUIDs for `Instance` and `Binding`.  The issue is that the objects we're creating in code in this test don't go through the defaults since they don't go through the API server.  So, we need to fill the GUID fields in ourselves for this test and others like it.